core-compose/src/main/java/com/squareup/workflow/ui/compose/internal/ViewRegistries.kt

```diff
@@ -40,3 +41,25 @@ import com.squareup.workflow.ui.ViewRegistry
   val viewFactory = remember(renderingType) { getFactoryFor(renderingType) }
   viewFactory.showRendering(rendering, hints, modifier)
 }
+
+/**
+ * Applies [transform] to each [ViewFactory] in this registry. Transformations are applied lazily,
+ * at the time of lookup via [ViewRegistry.getFactoryFor].
+ */
+internal fun ViewRegistry.mapFactories(
+  transform: (ViewFactory<*>) -> ViewFactory<*>
+): ViewRegistry = object : ViewRegistry {
+  override val keys: Set<KClass<*>> get() = this@mapFactories.keys
+
+  override fun <RenderingT : Any> getFactoryFor(
+    renderingType: KClass<out RenderingT>
+  ): ViewFactory<RenderingT> {
+    val transformedFactory = transform(this@mapFactories.getFactoryFor(renderingType))
+    check(transformedFactory.type == renderingType) {
+      "Expected transform to return a ViewFactory that is compatible with $renderingType, " +
+          "but got one with type ${transformedFactory.type}"
+    }
+    @Suppress("UNCHECKED_CAST")
+    return transformedFactory as ViewFactory<RenderingT>
+  }
+}
```

Comment:
This is the key to the decoupled implementation. This technique can be used to do other stuff too, like e.g. apply a modifier to each ViewFactory's view.

Reply:
I think I've already had a few use cases for this, should we move it to the main repo?

Reply:
Sure!